### ISSUE TITLE
Grid implementation test failing

### ISSUE THREAD

**Issue by egpbos** [amusecode/amuse#258]
Both on Travis and locally on macOS the core_tests/test_grid_implementation.py test fails and subsequently hangs for several minutes with output:

```sh
test1 (test.core_tests.test_grid_implementation.TestCImplementationInterface) ... building... done
-------------------------------------------------------
Child job 2 terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
-------------------------------------------------------
```

After some minutes it prints:

```sh
--------------------------------------------------------------------------
A request has timed out and will therefore fail:

  Operation:  LOOKUP: orted/pmix/pmix_server_pub.c:345

Your job may terminate as a result of this problem. You may want to
adjust the MCA parameter pmix_server_max_wait and try again. If this
occurred during a connect/accept operation, you can adjust that time
using the pmix_base_exchange_timeout parameter.
--------------------------------------------------------------------------
ERROR
```

and then continues with test2.

```sh
nosetests -sv test/core_tests/test_grid_implementation.py
```

Environment: the Travis build is on Linux (Ubuntu 14.04.5) using some gcc version, while the macOS High Sierra build uses clang 3.9, so it seems like this is a platform independent problem. Possibly it is an OpenMPI issue, but this also seems unlikely, since on Travis we use version 1.6.5-8 and on macOS we were using version 3.

Related to #255, https://github.com/amusecode/amuse/pull/244#issuecomment-400227040

**Comment by egpbos** [comment]
On my local macOS, the following test produces the exact same problem:
```
nosetests -sv test/reports/test_speed.py
```

**Comment by egpbos** [comment]
Same for `test/codes_tests/test_brutus.py`.

**Comment by egpbos** [comment]
Same for `test/codes_tests/test_c_implementation.py`.

**Comment by ipelupessy** [comment]
except for the brutus test these dynamically generate an executable, which is tested...

**Comment by egpbos** [comment]
Another test that fails in a similar but not completely equal way is `test8 (test.codes_tests.test_encounters.TestHandleEncounter)` which fails with output:

```sh
--------------------------------------------------------------------------
WARNING: Open MPI failed to TCP connect to a peer MPI process.  This
should not happen.

Your Open MPI job may now fail.

  Local host: ESLT0072
  PID:        39616
  Message:    connect() to 145.100.116.224:1031 failed
  Error:      Operation now in progress (36)
--------------------------------------------------------------------------
[ESLT0072:39619] 1 more process has sent help message help-mpi-btl-tcp.txt / client connect fail
```

and then hangs indefinitely (as opposed to the above mentioned hangs which eventually timeout and then nosetests can continue).

**Comment by ipelupessy** [comment]
original fixed by #272 
other issues opened
